src/core/cgroup.c

```diff
@@ -215,6 +216,9 @@ void cgroup_context_done(CGroupContext *c) {
         c->ip_filters_ingress = strv_free(c->ip_filters_ingress);
         c->ip_filters_egress = strv_free(c->ip_filters_egress);
 
+        while (c->pinned_bpf_progs)
```

Comment:
LIST_FOREACH_SAFE ?

Reply:
Did you mean `s/LIST_FOREACH/LIST_FOREACH_SAFE` in `cgroup_context_dump` ? This loop is responsible for freeing pointers from list tail to head, it is also consistent with other list fields, e.g.:

```
while (c->io_device_weights)
                cgroup_context_free_io_device_weight(c, c->io_device_weights);
```

Reply:
ok, looks good as-is then